File: jvm/src/main/kotlin/org/faziodev/saywhatstarter/types/Request.kt
package org.faziodev.saywhatstarter.types

data class Request(
    val responseId: String,
    val session: String,
    val queryResult: QueryResult?,
    val originalDetectIntentRequest: Map<String, Any>?
) {
    fun getContextParameter(contextName: String, parameterName: String): String =
        this.queryResult?.outputContexts?.firstOrNull { it.name == "$session/contexts/$contextName"}?.parameters?.get(parameterName) ?: ""
}

File: csharp/SayWhatStarterWebhook/Models/Request.cs
using System.Collections.Generic;
using System.Linq;

namespace SayWhatStarterWebhook.Models
{
    public class Request
    {
        public string ResponseId { get; set; }
        public string Session { get; set; }
        public QueryResult QueryResult { get; set; }
        public Dictionary<string, object> OriginalDetectIntentRequest { get; set; }

        public string GetContextParameter(string contextName, string parameterName)
        {
            return this.QueryResult?.OutputContexts
                ?.FirstOrDefault(c => c.Name?.Equals($"{this.Session}/contexts/{contextName}") ?? false)
                ?.Parameters?[parameterName];
        }
    }
}